File: problem/Min_Removals/minRemoval.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var minRemoval = function(nums, k) {
    nums.sort((a, b) => a - b);
    let i = 0;
    let maxLen = 0;
    
    for (let j = 0; j < nums.length; j++) {
        while (nums[j] > nums[i] * k) {
            i++;
        }
        maxLen = Math.max(maxLen, j - i + 1);
    }
    
    return nums.length - maxLen;
};

File: problem/Min_Removals/minRemoval.c
int cmp(const void *a, const void *b)
{
    return (*(int *)a - *(int *)b);
}

int minRemoval(int *nums, int numsSize, int k)
{
    qsort(nums, numsSize, sizeof(int), cmp);
    int i = 0;
    int maxLen = 0;

    for (int j = 0; j < numsSize; j++)
    {
        while ((long long)nums[j] > (long long)nums[i] * k)
        {
            i++;
        }
        int currentLen = j - i + 1;
        if (currentLen > maxLen)
        {
            maxLen = currentLen;
        }
    }

    return numsSize - maxLen;
}